corehq/apps/data_dictionary/models.py

```diff
@@ -8,18 +9,23 @@
 
 from corehq.apps.case_importer import exceptions
 
+property_type = namedtuple('property_type', 'slug display')
 
-PROPERTY_TYPE_CHOICES = (
-    ('date', _('Date')),
-    ('plain', _('Plain')),
-    ('number', _('Number')),
-    ('select', _('Multiple Choice')),
-    ('barcode', _('Barcode')),
-    ('gps', _('GPS')),
-    ('phone_number', _('Phone Number')),
-    ('password', _('Password')),
-    ('', 'No Type Currently Selected')
-)
+
+class PROPERTY_TYPES:
+    DATE = property_type('date', _('Date'))
+    PLAIN = property_type('plain', _('Plain'))
+    NUMBER = property_type('number', _('Number'))
+    SELECT = property_type('select', _('Multiple Choice'))
+    BARCODE = property_type('barcode', _('Barcode'))
+    GPS = property_type('gps', _('GPS'))
+    PHONE_NUMBER = property_type('phone_number', _('Phone Number'))
+    PASSWORD = property_type('password', _('Password'))
+    UNDEFINED = property_type('', _('No Type Currently Selected'))
+
+    @classmethod
+    def get_all(cls):
```

Comment:
if you make this class an `enum` you don't need this method since enums are iterable

Reply:
I did that at first, but it makes accessing it a little awkward.  The two options I considered there were:
```python
class PROPERTY_TYPES(Enum):
    gps = _('GPS')  # the field name doubles as the slug, so it needs to be lowercase

> PROPERTY_TYPES.gps.name
'gps'
> PROPERTY_TYPES.gps.value
'GPS'
```
I don't love this option, as `name` and `value` aren't really descriptive, and the "name" for undefined is `""`, so I'd have to work around that, I expect.

```python
class PROPERTY_TYPES(Enum):
    GPS = property_type('gps', _('GPS'))

> PROPERTY_TYPES.GPS.value.slug
'gps'
> PROPERTY_TYPES.GPS.value.display
'GPS'
```

Farid pointed me to https://docs.djangoproject.com/en/4.0/ref/models/fields/#field-choices-enum-types, which seems appropriate here, though access is the same as in my first example above.  It's not available in our version of Django, but I could be convinced to switch to that first option if that's a pattern we should expect to become familiar with.